app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.java

```diff
@@ -48,13 +59,33 @@ protected void onCreate(Bundle savedInstanceState) {
         init();
     }
 
+    private final ControllerListener loadingListener = new BaseControllerListener<ImageInfo>() {
+        @Override
+        public void onSubmit(String id, Object callerContext) {
+        }
+
+        @Override
+        public void onIntermediateImageSet(String id, @Nullable ImageInfo imageInfo) {
```

Comment:
Sometimes intermediate image is also slowly appears. I think we should also use the progressbar for that phase. What do you think @kbhardwaj123 ?

Reply:
@neslihanturan actually i inspected this thing and found that if i place a log statement for `onIntermediateImageSet` it won't appear in the logs, i think it would be better if i make the Indeterminate spinner INVISIBLE as soon as Intermediate image is set